configure

```diff
@@ -292,277 +332,749 @@ LIBS="${LIBS} mpfr"
 fi
 CONFIG_GC="#define FLINT_USES_GC ${WANT_GC}"
 
 # defaults for CC, CXX and AR
 
 if [ -z "$CC" ]; then
-   CC=gcc
+    CC=gcc
 fi
 
 if [ -z "$CXX" ]; then
-   CXX=g++
+    CXX=g++
 fi
 
 if [ -z "$AR" ]; then
-   AR=ar
+    AR=ar
 fi
 
 #handle gc and reentrant flags
 
 if [ "$WANT_GC" = "1" ]; then
-      TLS=0
-      if [ "$WANT_TLS" = "1" ]; then
-          echo "****WARNING**** GC does not support TLS....disabling TLS"
-	  echo "GC does not support TLS....disabling TLS" >> config.log
-      fi
-      cp fmpz/link/fmpz_gc.c fmpz/fmpz.c
-      cp fmpz-conversions-gc.in fmpz-conversions.h
+    TLS=0
+    if [ "$WANT_TLS" = "1" ]; then
+        echo "****WARNING**** GC does not support TLS....disabling TLS"
+        echo "GC does not support TLS....disabling TLS" >> config.log
+    fi
+    cp fmpz/link/fmpz_gc.c fmpz/fmpz.c
+    cp fmpz-conversions-gc.in fmpz-conversions.h
 else
-   if [ "$REENTRANT" = "1" ]; then
-      cp fmpz/link/fmpz_reentrant.c fmpz/fmpz.c
-      cp fmpz-conversions-reentrant.in fmpz-conversions.h
-   else
-      cp fmpz/link/fmpz_single.c fmpz/fmpz.c
-      cp fmpz-conversions-single.in fmpz-conversions.h
-   fi
+    if [ "$REENTRANT" = "1" ]; then
+        cp fmpz/link/fmpz_reentrant.c fmpz/fmpz.c
+        cp fmpz-conversions-reentrant.in fmpz-conversions.h
+    else
+        cp fmpz/link/fmpz_single.c fmpz/fmpz.c
+        cp fmpz-conversions-single.in fmpz-conversions.h
+    fi
 fi
-# Architecture handler
 
-KERNEL=`uname`
+################################################################################
+################################################################################
+# build handler
+################################################################################
+################################################################################
+
+BUILD_KERNEL=`uname`
 
+echo "Checking build..."
+echo "Checking build..." >> config.log
 if [ -z "$BUILD" ]; then
-   ARCH=`uname -m`
-
-   if [ "$(uname | cut -d_ -f1)" = "MSYS" ]; then
-      if [ "$ARCH" = "x86_64" ]; then
-         OS="MINGW64"
-      else
-         OS="MINGW32"
-      fi
-   elif [ "$(uname | cut -d_ -f1)" = "MINGW32" ]; then
-      if [ "$ABI" = "64" ]; then
-         OS="MINGW64"
-      else
-         OS="MINGW32"
-      fi 
-   elif [ "$(uname | cut -d_ -f1)" = "MINGW64" ]; then
-      if [ "$ABI" = "32" ]; then
-         OS="MINGW32"
-      else
-         OS="MINGW64"
-      fi
-   elif [ "$(uname | cut -d_ -f1)" = "CYGWIN" ]; then
-      if [ "$ARCH" = "x86_64" ]; then
-         if [ "$ABI" = "32" ]; then
-            OS="CYGWIN32"
-         else
-            OS="CYGWIN64"
-            ABI="64"
-         fi
-      else
-         OS="CYGWIN32"
-      fi
-   else
-      OS=`uname -s`
-   fi
+    # Build was not specified, so try to guess it
+
+    # Set MARCH_FLAG to -march=native -mno-avx512f, where the second option is
+    # to disable AVX512 on those systems that supports it.
+    MARCH_FLAG="-march=native -mno-avx512f"
```

Comment:
Hmm...
```
gcc-11: error: unrecognized command-line option '-mno-avx512f'
```
I don't think it's a good idea to pass that on non-intel machines, nor to compilers which are too old (or perhaps too new as well?) or which simply are not GCC (or clang). By forcing these flags there, compilation with different compilers and on different architectures gets that much harder "out of the box" :-(

Reply:
Good point.